class.jetpack.php

```diff
@@ -738,11 +732,65 @@ function( $callable ) {
 		add_filter( 'jetpack_token_request_body', array( __CLASS__, 'filter_token_request_body' ) );
 	}
 
+	/**
+	 * Runs on plugins_loaded but with a higher priority to make sure we're initializing
+	 * before other hooks.
+	 *
+	 * @action plugins_loaded
+	 * @priority 5
+	 */
+	public function early_initalization() {
+		/*
+		 * Enable enhanced handling of previewing sites in Calypso
+		 */
+		if ( self::is_active() ) {
+			require_once JETPACK__PLUGIN_DIR . '_inc/lib/class.jetpack-iframe-embed.php';
+			add_action( 'init', array( 'Jetpack_Iframe_Embed', 'init' ), 9, 0 );
+			require_once JETPACK__PLUGIN_DIR . '_inc/lib/class.jetpack-keyring-service-helper.php';
+			add_action( 'init', array( 'Jetpack_Keyring_Service_Helper', 'init' ), 9, 0 );
+		}
+
+		Sync_Main::init();
+
+		// Check for WooCommerce support.
+		Sync_Actions::initialize_woocommerce();
+
+		// Check for WP Super Cache.
+		Sync_Actions::initialize_wp_super_cache();
+	}
+
+	/**
+	 * Runs on plugins_loaded but with a lower priority.
```

Comment:
> lower priority.

This always makes things confusing 😉 While `90` is lower priority, it's a higher number, so what executes first? Maybe let's change this to "executed later" or "executed after the plugins have been loaded" to improve clarity.